src/NHibernate.Test/NHSpecificTest/NH3850/Fixture.cs

```diff
@@ -0,0 +1,1314 @@
+				Assert.AreEqual(_searchName1, result.Name);
+				Assert.IsInstanceOf<DomainClassCExtendedByD>(result);
+			}
+		}
+
+		// Non-reg case
+		[Test]
+		public void SingleOrDefaultE()
+		{
+			using (var session = OpenSession())
+			{
+				var query = session.Query<DomainClassE>();
+				DomainClassE result = null;
+				Assert.Throws<InvalidOperationException>(() => { result = query.SingleOrDefault(); });
+			}
+		}
+
+		// Non-reg case
+		[Test]
+		public void SingleOrDefaultEWithName()
+		{
+			using (var session = OpenSession())
+			{
+				var query = session.Query<DomainClassE>();
+				DomainClassE result = null;
+				Assert.DoesNotThrow(() => { result = query.SingleOrDefault(dc => dc.Name == _searchName1); });
+				Assert.IsNotNull(result);
+				Assert.AreEqual(_searchName1, result.Name);
+				Assert.IsInstanceOf<DomainClassE>(result);
+			}
+		}
+
+		// Non-reg case
+		[Test]
+		public void SingleOrDefaultF()
+		{
+			using (var session = OpenSession())
+			{
+				var query = session.Query<DomainClassF>();
+				DomainClassF result = null;
+				Assert.DoesNotThrow(() => { result = query.SingleOrDefault(); });
+				Assert.IsNull(result);
+			}
+		}
+
+		// Non-reg case
+		[Test]
+		public void SingleOrDefaultFWithName()
+		{
+			using (var session = OpenSession())
+			{
+				var query = session.Query<DomainClassF>();
+				DomainClassF result = null;
+				Assert.DoesNotThrow(() => { result = query.SingleOrDefault(dc => dc.Name == _searchName1); });
+				Assert.IsNull(result);
+			}
+		}
+
+		// Non-reg case
+		[Test]
+		public void SingleOrDefaultGBase()
+		{
+			using (var session = OpenSession())
+			{
+				var query = session.Query<DomainClassGExtendedByH>();
+				DomainClassGExtendedByH result = null;
+				Assert.Throws<InvalidOperationException>(() => { result = query.SingleOrDefault(); });
+			}
+		}
+
+		// Non-reg case
+		[Test]
+		public void SingleOrDefaultGBaseWithName()
+		{
+			using (var session = OpenSession())
+			{
+				var query = session.Query<DomainClassGExtendedByH>();
+				DomainClassGExtendedByH result = null;
+				Assert.Throws<InvalidOperationException>(() => { result = query.SingleOrDefault(dc => dc.Name == _searchName1); });
+			}
+		}
+
+		// Non-reg case
+		[Test]
+		public void SingleOrDefaultObject()
+		{
+			using (var session = OpenSession())
+			{
+				var query = session.Query<object>();
+				object result = null;
+				Assert.Throws<InvalidOperationException>(() => { result = query.SingleOrDefault(); });
+			}
+		}
+
+		// Failing case till NH-3850 is fixed
+		[Test]
+		public void SumBBase()
+		{
+			Sum<DomainClassBExtendedByA>(3);
+		}
+
+		// Failing case till NH-3850 is fixed
+		[Test]
+		public void SumCBase()
+		{
+			Sum<DomainClassCExtendedByD>(3);
+		}
+
+		// Non-reg case
+		[Test]
+		public void SumE()
+		{
+			Sum<DomainClassE>(3);
+		}
+
+		// Non-reg case
+		[Test, Ignore("Candidate potential breaking changes: Sum would now yield 0 instead of null on empty result sets.")]
```

Comment:
Rebase & conflict resolution miss, I had removed it prior to rebase. Going to remove it again.